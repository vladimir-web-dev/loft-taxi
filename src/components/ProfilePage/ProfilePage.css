.profile {   
    background-size: cover;
    background-position: center;
    background-image: 
        linear-gradient(rgba(0,0,0, .3), rgba(0,0,0, .4)),
        url('../../images/background.png');
}