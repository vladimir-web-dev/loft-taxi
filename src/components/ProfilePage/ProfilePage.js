import React from 'react';
import Header from '../Header';
import PropTypes from 'prop-types';
import { authorizationCheckHOC } from '../../HOCs'

let ProfilePage = (props) => {
    return (
        <section className='section'>
            <Header changePage={props.changePage} />
            <h1>Profile</h1>
        </section>
    );
}

ProfilePage = authorizationCheckHOC(ProfilePage);

ProfilePage.propTypes = {
    changePage: PropTypes.func.isRequired
}


export default ProfilePage;