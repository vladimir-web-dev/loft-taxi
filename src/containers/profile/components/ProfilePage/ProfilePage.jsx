import React from "react";
import { Header } from "../../../general/Header";
import { ProfileForm } from "../ProfileForm";
import { ProfileFormSaved } from "../ProfileFormSaved";
import { useSelector } from "react-redux";
import {
  getIsUpdatedSelector,
  getIsFetchingSelector,
  getIsUpdatingSelector
} from "../../../profile/store";
import containerStyles from "../../../../css_modules/Container.module.css";
import { LoadingModal } from "../../../general/LoadingModal";

export function ProfilePage() {
  const isFetching = useSelector(state =>
    getIsFetchingSelector(state.profileReducer)
  );

  const isUpdating = useSelector(state =>
    getIsUpdatingSelector(state.profileReducer)
  );

  const isUpdated = useSelector(state =>
    getIsUpdatedSelector(state.profileReducer)
  );

  return (
    <>
      <section className={containerStyles.vertical}>
        <Header />
        <div className={containerStyles.formContainer}>
          <div className={containerStyles.profileForm}>
            {isUpdated ? <ProfileFormSaved /> : <ProfileForm />}
          </div>
        </div>
      </section>
      <LoadingModal showModal={isFetching || isUpdating} />
    </>
  );
}
