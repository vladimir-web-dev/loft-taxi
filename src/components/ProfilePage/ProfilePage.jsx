import React from 'react';
import Header from '../Header';
import Grid from '@material-ui/core/Grid';
import { ProfileForm } from '../ProfileForm';
import './ProfilePage.css';

const ProfilePage = (props) => {
    return (
        <section className='section profile'>
            <Header  />
            <Grid container justify='center'>
                <Grid item>
                    <ProfileForm />
                </Grid>
            </Grid>
        </section>
    );
}

export default ProfilePage;