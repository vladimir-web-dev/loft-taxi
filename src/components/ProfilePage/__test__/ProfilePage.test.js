import React from 'react';
import ReactDOM from 'react-dom';
import ProfilePage from '../ProfilePage';
import {render} from '@testing-library/react';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ProfilePage changePage={() => {}}></ProfilePage>, div);
    ReactDOM.unmountComponentAtNode(div);
})

it('matches snapshot', () => {
    const {container} = render(<ProfilePage changePage={() => {}}></ProfilePage>)
    expect(container).toMatchSnapshot(); 
})