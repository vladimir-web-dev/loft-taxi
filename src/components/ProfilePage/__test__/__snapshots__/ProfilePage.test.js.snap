// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <h1>
    You are unauthorized to enter this page
  </h1>
</div>
`;
